framework/src/kernels/ArrayKernel.C

```diff
@@ -137,9 +138,10 @@ ArrayKernel::computeJacobian()
     for (_i = 0; _i < _test.size(); _i++)
       for (_j = 0; _j < _phi.size(); _j++)
       {
-        RealEigenVector v = _JxW[_qp] * _coord[_qp] * computeQpJacobian();
+        _work_vector = computeQpJacobian();
+        _work_vector *= _JxW[_qp] * _coord[_qp];
```

Comment:
is this split required? (just curious)

Reply:
I noticed the assignment operation was significantly faster doing this.

Reply:
The order of multiplications matters, so the compiler cannot just arbitrarily reorder it, but I wonder if 

```
RealEigenVector v = computeQpJacobian() * _JxW[_qp] * _coord[_qp];
```

would word the same way as your change.

Reply:
Hmm, that's a good point. I'll give that a shot.

Reply:
Looks like the order of operation doesn't matter for this case. I get the same results with `computeQpResidual() * _JxW[_qp] * _coord[_qp];` and `_JxW[_qp] * _coord[_qp] * computeQpResidual();`. Separating them out is faster.

Reply:
Wait! Nevermind, looks like it does matter.

Reply:
Okay changing the order is basically the same as separating it out. I'll change accordingly.

Reply:
Done. 4 less lines of code woo!

Reply:
Are we sure about this? If the evaluation is done as `(computeQpJacobian() * _JxW[_qp]) * _coord[_qp]`, we are wasting some operation than `computeQpJacobian() * (_JxW[_qp] * _coord[_qp])`, am I right? The default seems the first one, i.e. the evaluation is done from left to right.

Reply:
> evaluation is done from left to right.

I would not rely on this. Compilers are magical beings.

Reply:
No, IEEE 754 determines the order of multiplications. The order of _evaluation_ is up to the compiler in this case, because the `*` operator is not a _sequence point_